staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,297 @@
+	// `type` is the type of the condition.
+	Type PriorityLevelConfigurationConditionType `json:"type,omitempty" protobuf:"bytes,1,opt,name=type"`
+	// `status` is the status of the condition.
+	// Can be True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status,omitempty" protobuf:"bytes,2,opt,name=status"`
+	// `lastTransitionTime` is the last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,3,opt,name=lastTransitionTime"`
+	// `reason` is a unique, one-word, CamelCase reason for the condition's last transition.
+	Reason string `json:"reason,omitempty" protobuf:"bytes,4,opt,name=reason"`
+	// `message` is a human-readable message indicating details about last transition.
+	Message string `json:"message,omitempty" protobuf:"bytes,5,opt,name=message"`
+}
+
+// Subject matches a set of users.
+// Syntactically, Subject is a general API object reference.
+// Authorization produces a username and a set of groups, and we imagine special kinds of non-namespaced objects,
+// User and Group in API group "flowcontrol.apiserver.k8s.io", to represent such a username or group.
+// The only kind of true object reference that currently will match any users is ServiceAccount.
+type Subject struct {
+	// `kind` of object being referenced. Values defined by this API group are "User", "Group", and "ServiceAccount".
+	// If the kind value is not recognized, the flow-control layer in api-server should report an error.
+	Kind string `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+	// `apiGroup` holds the API group of the referenced subject.
+	// Defaults to "" for ServiceAccount subjects.
+	// Defaults to "flowcontrol.apiserver.k8s.io" for User and Group subjects.
+	// +optional
+	APIGroup string `json:"apiGroup,omitempty" protobuf:"bytes,2,opt.name=apiGroup"`
+	// `name` of the object being referenced.  To match regardless of name, use NameAll.
+	// Required.
+	Name string `json:"name" protobuf:"bytes,3,opt,name=name"`
+	// `namespace` of the referenced object.  If the object kind is non-namespace, such as "User" or "Group", and this value is not empty
+	// the Authorizer should report an error.
+	// +optional
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,4,opt,name=namespace"`
+}
+
+// PolicyRule holds information that describes a policy rule, but does not contain information
+// about who the rule applies to or which namespace the rule applies to.
+type PolicyRule struct {
+	// `verbs` is a list of Verbs that apply to ALL the ResourceKinds and AttributeRestrictions contained in this rule.
+	// VerbAll represents all kinds.
+	Verbs []string `json:"verbs" protobuf:"bytes,1,rep,name=verbs"`
+	// `apiGroups` is the name of the APIGroup that contains the resources.  If multiple API groups are specified, any action requested against one of
+	// the enumerated resources in any API group will be allowed. APIGroupAll represents all api groups.
```

Comment:
Say `"*"` instead of APIGroupAll, it's not obvious you're referring to the constant and I'm not sure if that's presented in the documentation well. Applies to all these fields.

Reply:
fixed in #81788

Reply:
I think Daniel's point is that the corresponding change should also be made to all the fields here.